### PR TITLE
libwebsocket: fix compile with mbedtls and update to 4.3.3

### PR DESCRIPTION
**Pull request by orangepizza**
upstreamed patches was removed in process, and add dependency for MBEDTLS_VERSION_C as this package needs that to compile when used mbedtls as backend. (not sure how it worked before?)
mbedtls 3.6 moved one of function used in this into private header, patch redefined it to be able to compiled. a more detailed fix will be needed form upstream, but it should be enough for us.

### CODE REVIEW

**Comment by orangepizza** [comment]
It'd needed patch anyway because it has silent dependent on mbedtls_version_c but I think I can patch it to use mecro string instead, removing that

**Comment by orangepizza** [comment]
this commit already fixed reason it caused error: not really needed to disable werror
this got merged upstream, but upstream doesn't look like it'd have frequent patch releases
 https://github.com/warmcat/libwebsockets/commit/1e18a654d9d6b0a5f35de65fc6797592f85d197d

**Comment by dddaniel** [comment]
> this commit already fixed reason it caused error: not really needed to disable werror this got merged upstream, but upstream doesn't look like it'd have frequent patch releases [warmcat/libwebsockets@1e18a65](https://github.com/warmcat/libwebsockets/commit/1e18a654d9d6b0a5f35de65fc6797592f85d197d)

Do programs that link against libwebsockets compile successfully with your commit ?
I had issues with the upstream main branch of libwebsockets and your backported commit.
Libwebsockets itself compiles, but linking against it is giving me this:

/usr/bin/ld: /usr/local/lib/libwebsockets.so: undefined reference to `private_mbedtls_x509_get_name'

**Comment by orangepizza** [comment]
reverted to my original fix, this surely works with something linked with it complied

**Comment by neheb** [comment]
Squash these commits. Also, don't use .diff files.

**Comment by orangepizza** [comment]
did that, but doesn't make refresh strips any patch headers, making it a diff?

**Comment by neheb** [comment]
quilt does not strip patch headers.

**Comment by neheb** [comment]
Run

‘’’
make package/libwebsockets/{clean,refresh}
‘’’

File: libs/libwebsockets/Makefile [hunk 1]
```diff
@@ -61,7 +61,7 @@ endef
 define Package/libwebsockets-mbedtls
 	$(call Package/$(PKG_NAME)/Default)
 	TITLE += (mbedTLS)
-	DEPENDS += +libmbedtls
+	DEPENDS += +libmbedtls @MBEDTLS_VERSION_C
```
Comment (hauke) [inline libs/libwebsockets/Makefile]: We should probably activate this option in the main repository by default. Could you create a PR for that too please.

**Pull request merged**